sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java

```diff
@@ -235,10 +236,8 @@ public ConfigurationClientBuilder addPolicy(HttpPipelinePolicy policy) {
      *
      * @param client The HTTP client to use for requests.
      * @return The updated ConfigurationClientBuilder object.
-     * @throws NullPointerException If {@code client} is {@code null}.
      */
     public ConfigurationClientBuilder httpClient(HttpClient client) {
-        Objects.requireNonNull(client);
```

Comment:
What's the reason of removing the null checking?

Reply:
As discussed with @JonathanGiles it isn't likely the right choice to throw an exception here as having a null HttpClient is valid for the builder.